### PR TITLE
[SHELL32][SYSSETUP] Shell folders in My Documents

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

If the product type is `NtProductWinNt`, then some special folders should be in My Documents.
JIRA issue: [CORE-13795](https://jira.reactos.org/browse/CORE-13795)

- Implement `CSIDL_Type_InMyDocuments` `CSIDL` type.
- Add "Product Options" Wizard Page into ReactOS Setup.

### CODE REVIEW

File: dll/win32/shell32/wine/shellpath.c [hunk 1]
```diff
@@ -1527,6 +1528,30 @@ BOOL _SHGetUserProfileDirectoryW(HANDLE hToken, LPWSTR szPath, LPDWORD lpcchPath
...
+    error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, szProductOptions, 0, KEY_READ, &hKey);
+    if (error != ERROR_SUCCESS)
+        return ret;
+
+    cbValue = sizeof(szValue);
+    error = RegQueryValueExW(hKey, L"ProductType", NULL, NULL, (LPBYTE)szValue, &cbValue);
+    if (error != ERROR_SUCCESS)
+        return ret;
+
+    ret = (lstrcmpiW(szValue, L"ServerNT") == 0);
```
Comment (HBelusca) [inline dll/win32/shell32/wine/shellpath.c]: Note: using `RtlGetNtProductType()` (see https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/rtl/version.c;hb=4e964c7cb49936036607b93a79cc2c55a18ca0c7#l71 ) could work as well.

**Comment by ale5000-git** [comment]
@katahiromz

> If the product type is not server, then some special folders should be in My Documents.

Who said that?

On Windows Vista/7/8/10 (home versions) special folders are NOT inside Documents.

**Comment by HBelusca** [comment]
> Who said that?
> 
> On Windows Vista/7/8/10 (home versions) special folders are NOT inside Documents.
> 
> **PS: This was nice => #1521**

That's what happens on Windows <= 2003 (incl. XP). But yes the PR #1521 was a better thing, I think...

File: dll/win32/shell32/wine/shellpath.c [hunk 2]
```diff
@@ -648,6 +648,7 @@ static const WCHAR AllUsersW[] = {'P','u','b','l','i','c',0};
 
 typedef enum _CSIDL_Type {
     CSIDL_Type_User,
+    CSIDL_Type_InMyDocuments,
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: OK.

File: dll/win32/shell32/wine/shellpath.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/shell32/wine/shellpath.c]: OK.

**Comment by katahiromz** [comment]
syssetup
Coming soon in `syssetup`.

**Comment by katahiromz** [comment]
after-1

**Comment by katahiromz** [comment]
after-2
`RtlGetNtProductType` won't return a correct value.

**Comment by katahiromz** [comment]
after
Umm...

**Comment by katahiromz** [comment]
after

**Comment by katahiromz** [comment]
File `hivesys.inf` seems writing registry key `PlatformOptions`...

File: dll/win32/syssetup/wizard.c [hunk 3]
```diff
@@ -355,11 +355,170 @@ AckPageDlgProc(HWND hwndDlg,
...
+            pSetupData = (PSETUPDATA)((LPPROPSHEETPAGE)lParam)->lParam;
+            SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)pSetupData);
+
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_ADDSTRING, 0, (LPARAM)L"2003 Server (Default)");
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_ADDSTRING, 0, (LPARAM)L"XP");
+
+            SendDlgItemMessageW(hwndDlg, IDC_PLATFORM_OPTIONS, CB_SETCURSEL, PLATFORM_OPTION_2003_SERVER, 0);
+            OnChoose2003Server(hwndDlg);
+        }
+        break;
```
Comment (Fraizeraust) [inline dll/win32/syssetup/wizard.c]: `WM_INITDIALOG` returns FALSE/TRUE upon finishing the control dialog initialization. Why break it?

File: dll/win32/syssetup/wizard.c [hunk 4]
```diff
@@ -355,11 +355,170 @@ AckPageDlgProc(HWND hwndDlg,
...
+    //
+    //        /* write ProductType */
+    //        pData = L"WinNT";
+    //        cbData = (lstrlenW(pData) + 1) * sizeof(WCHAR);
+    //        error = RegSetValueExW(hKey, L"ProductType", 0, REG_SZ, (BYTE *)pData, cbData);
+    //        break;
+    //}
+    //
+    //RegCloseKey(hKey);
+    //return error == ERROR_SUCCESS;
```
Comment (Fraizeraust) [inline dll/win32/syssetup/wizard.c]: Nothing wrong here but next time use `#if 0 {...}` to disable a certain chunk of code to be compiled -- FIY.

**Comment by katahiromz** [comment]
after
Improving...

**Comment by katahiromz** [comment]
after

**Comment by katahiromz** [comment]
VirtualBox_ReactOS_10_02_2020_00_12_47

**Comment by katahiromz** [comment]
Can you provide the descriptions?
Is the icon OK?

File: dll/win32/syssetup/wizard.c (hunk 3 above)
Comment (katahiromz) [reply dll/win32/syssetup/wizard.c]: OK.

File: dll/win32/syssetup/wizard.c (hunk 4 above)
Comment (katahiromz) [reply dll/win32/syssetup/wizard.c]: Uncomment and enabled.

**Comment by katahiromz** [comment]
after

**Comment by katahiromz** [comment]
after

**Comment by HBelusca** [comment]
Before we forget, you should verify that the new installer page is compatible with unattended installation (and when the unattended installation is used, the ReactOS Server product mode is used).

**Comment by katahiromz** [comment]
https://reactos.org/wiki/Create_an_unattended_Installation_CD

**Comment by katahiromz** [comment]
unattended
Unattended installation is successful.

**Pull request merged**